### PR TITLE
Expanding documentation

### PR DESCRIPTION
**Pull request by JustusAdam**
I looked through the documentation and I had some ideas on how it could be improved.

I applied the changes to one class only for now, but if people like if I can expand that to the other classes.

This is really more of an RFC than a PR for now.

I've paid special attention to avoid explaining the purpose of a method with its name and use synonyms instead.

### CODE REVIEW

**Comment by bocharsky-bw** [comment]
I like it! I'll be very glad if you improved an other classes too :+1:

File: src/MutableArray.php [hunk 1]
```diff
@@ -186,11 +191,11 @@ public function replaceIn(array $array, $recursively = false)
     }
 
     /**
-     * Combine array values used as keys with a given array values
+     * Create an array using this array as keys and the otehr array as values.
```
Comment (bocharsky-bw) [inline src/MutableArray.php]: Fix typo in `other`

File: src/MutableArray.php [hunk 2]
```diff
@@ -200,11 +205,11 @@ public function combineWith(array $array)
     }
 
     /**
-     * Combine array values to a given array values used as keys
+     * Create an array using this array as values and the otehr array as keys.
```
Comment (bocharsky-bw) [inline src/MutableArray.php]: Fix typo in `other`

**Comment by bocharsky-bw** [comment]
Nice! I just add some tiny comments.

Could I merge it or you want to change more files in this PR?

**Comment by JustusAdam** [comment]
I did plan on adding more files, so we can review them here.

**Comment by bocharsky-bw** [comment]
OK, just tell me when you finished and will be able to merge.

**Comment by JustusAdam** [comment]
So I expanded the the documentation to `ImmutableArray` and `AbstractArray` and I am pretty pleased with the result.

I searched my changes for mistakes multiple times but I may have missed something. You can/should look it over just to be sure it's all good. 

But as far as I'm concerned you can now merge it now.

**Comment by JustusAdam** [comment]
I also added the filterNot function which doesn't do too much but I thought it might come in handy.

File: src/AbstractArray.php [hunk 3]
```diff
@@ -216,30 +220,44 @@ public function __construct(array $elements = [])
...
 
     /**
-     * Apply the given function to every array element
+     * Filter the array for elements not satisfying the predicate $func.
+     *
+     * @param Closure $func predicate
+     *
+     * @return $this The same array with only element not satisfying $func
+     */
+    public function filterNot(Closure $func)
```
Comment (bocharsky-bw) [inline src/AbstractArray.php]: You better should be added it in a separate PR. But open an issue for it first would be the best way.

Anyway, I don't think we really need this method :-1: . We could easily use negation in a custom logic of closure.

So I prefer to get rid of it.

Comment (JustusAdam) [inline src/AbstractArray.php]: All right, no problem. Was just a thought.

File: src/AbstractArray.php [hunk 4]
```diff
@@ -396,11 +400,11 @@ public function isNumeric()
     }
 
     /**
-     * Check if the given value exists in array
+     * Check if the given value exists in the array
```
Comment (bocharsky-bw) [inline src/AbstractArray.php]: Seems we should add `.` to the end here too

File: src/AbstractArray.php [hunk 5]
```diff
@@ -462,7 +466,7 @@ public function getRandom()
     }
 
     /**
-     * Pick a given number of random keys/indexes out of array
+     * Pick a given number of random keys/indexes out of this array
```
Comment (bocharsky-bw) [inline src/AbstractArray.php]: The `.` at the end again

**Comment by bocharsky-bw** [comment]
I just merged this PR with rebase and squash your commits in https://github.com/bocharsky-bw/Arrayzy/commit/35b862a5f7a677b3eb0c259a020e1bf76fe1e4d2. All your work accepted.

Great work done! Thank you @JustusAdam !
